modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2210,6 +2211,8 @@ def apply_full_axis(
         keep_partitioning : boolean, default: True
             The flag to keep partition boundaries for Modin Frame.
             Setting it to True disables shuffling data from one partition to another.
+        synchronize : boolean, default: True
+            Synchronize external indexes (`new_index`, `new_columns`) with internal indexes.
```

Comment:
```suggestion
            Synchronize external indexes (`new_index`, `new_columns`) with internal indexes.
            This could be used when you're certain that the indices in partitions are equal to
            the provided hints in order to save time on syncing them.
```